src/Agoda.Analyzers/AgodaCustom/AG0043NoBuildServiceProvider.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Agoda.Analyzers.Helpers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Agoda.Analyzers.AgodaCustom
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AG0043NoBuildServiceProvider : DiagnosticAnalyzer
+    {
+        public const string DIAGNOSTIC_ID = "AG0043";
+
+        private static readonly LocalizableString Title = new LocalizableResourceString(
+            nameof(CustomRulesResources.AG0043Title),
+            CustomRulesResources.ResourceManager,
+            typeof(CustomRulesResources));
+
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(
+            nameof(CustomRulesResources.AG0043Title),
+            CustomRulesResources.ResourceManager,
+            typeof(CustomRulesResources));
+
+        private static readonly LocalizableString Description
+            = DescriptionContentLoader.GetAnalyzerDescription(nameof(AG0043NoBuildServiceProvider));
+
+        private static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            DIAGNOSTIC_ID,
+            Title,
+            MessageFormat,
+            AnalyzerCategory.CustomQualityRules,
+            DiagnosticSeverity.Error,
+            AnalyzerConstants.EnabledByDefault,
+            Description,
+            "https://github.com/agoda-com/AgodaAnalyzers/issues/190",
```

Comment:
lets do the documentation in the doc repo, not in issues, this will allow us to be more accurate, issues are chatty

Reply:
https://github.com/agoda-com/AgodaAnalyzers/pull/193

I did another MR to update docs, i think replicate this.

Ideally these should be in our standards, but things like this is more of a bug than a standard, so i think we need a new ay to store like this